app/uk/gov/hmrc/agentsubscription/service/CTReferenceService.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscription.service
+
+import javax.inject.{ Inject, Singleton }
+import play.api.Logger
+import uk.gov.hmrc.agentmtdidentifiers.model.Utr
+import uk.gov.hmrc.agentsubscription.connectors.DesConnector
+import uk.gov.hmrc.agentsubscription.model.Crn
+import uk.gov.hmrc.agentsubscription.service.CtUtrMatchResponse._
+import uk.gov.hmrc.http.{ BadRequestException, HeaderCarrier, NotFoundException }
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+sealed trait CtUtrMatchResponse
+
+object CtUtrMatchResponse {
+  case object Match extends CtUtrMatchResponse
+  case object NoMatch extends CtUtrMatchResponse
+  case object UtrNotFound extends CtUtrMatchResponse
+  case object InvalidCrn extends CtUtrMatchResponse
+  case object UnknownError extends CtUtrMatchResponse
+}
+
+@Singleton
+class CTReferenceService @Inject() (desConnector: DesConnector) {
+
+  def matchCorporationTaxUtrWithCrn(utr: Utr, crn: Crn)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[CtUtrMatchResponse] = {
```

Comment:
returning `CtUtrMatchResponse` from this service and dealing those responses in the controller is an overkiil , instead this method can simply return `Future[Boolean]`
method name could be `ctUtrMatchesCrn`  ?